src/pages/index.js

```diff
@@ -21,36 +37,121 @@ const imageModal = new ModalWithImage({
   modalSelector: "#view-image-modal",
 });
 
+const deleteModal = new ModalDelete({
+  modalSelector: "#delete-image-modal",
+  cardRemovalHandler: handleCardRemoval,
+  card: undefined,
+});
+
 // profile section
 
 const profileEl = new UserInfo({
   nameSelector: "#profile-name",
   descriptionSelector: "#profile-description",
+  avatarSelector: "#avatar",
 });
 
 // submission handlers
 
 function handleProfileEditSubmit(e, inputValues) {
   e.preventDefault();
+  consts.profileEditDiv.querySelector(".modal__button").innerText = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```